index.js

```diff
@@ -375,7 +375,7 @@ module.exports = function StateProvider(makeRenderer, rootElement, stateRouterOp
 		const parametersWereNotPassedIn = !parameters
 
 		return stateNameMatches
-			&& (parametersWereNotPassedIn || Object.keys(parameters).every(key => parameters[key] === currentState.parameters[key]))
+			&& (parametersWereNotPassedIn || Object.keys(parameters).every(key => parameters[key]?.toString() === currentState.parameters[key]))
```

Comment:
```suggestion
			&& (parametersWereNotPassedIn || Object.keys(parameters).every(key => parameters[key] + '' === currentState.parameters[key]))
```

Optional chaining might be a breaking change.

Optional chaining is pretty well supported.  It appeared in node v14.  I'm guessing ASR currently works with older node versions.  Ideally we'd run the tests against the oldest supported version of node, or document somehow what is supported.

Reply:
Speaking of which, is this part of the readme still true? https://github.com/TehShrike/abstract-state-router#browser-compatibility

Reply:
Oh, no it's not... I should update that

Reply:
All right, I switched to the concatenate-with-an-empty-string method for backwards compatibility